### PR TITLE
Collision funcs

### PR DESCRIPTION
**Pull request by m00nbyt3**

### CODE REVIEW

File: src/collision/collision.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/*   collision.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ycarro <ycarro@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/23 14:59:18 by ycarro            #+#    #+#             */
+/*   Updated: 2023/02/23 17:12:59 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <collision.h>
```
Comment (arikhativa) [inline src/collision/collision.c:13]: does this work? shouldn't we use :
`#include <collision.h> -> #include "collision.h"`

File: src/collision/collision.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+int	it_collides(t_ray *ray_data, char **map, t_point *collision)
```
Comment (arikhativa) [inline src/collision/collision.c:15]: boolean func should have a "is" prefix -> is_collides
and you can use the t_bool object that is defined in macro.h

File: src/collision/collision.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	collider.x = ray_data->x_pos.x / 100;
+	collider.y = ray_data->x_pos.y / 100;
+	if (map[collider.x][collider.y] == 1)
+		return (0);
+	collider.x = ray_data->y_pos.x / 100;
+	collider.y = ray_data->y_pos.y / 100;
+	if (map[collider.x][collider.y] == 1)
+		return (1);
+	return (-1);
```
Comment (arikhativa) [inline src/collision/collision.c:47]: i guess thie -1 if for errors?
if so then make sure it returns all the why up to main and there prints the error msg

**Review by arikhativa — CHANGES_REQUESTED** [review]

File: src/collision/collision.c (hunk 3 above)
Comment (m00nbyt3) [reply src/collision/collision.c:47]: The -1 here indicates there is no collision
0  -> Colisiion on the x ray step
1  -> Collision on the y ray step
-1 -> No collision

File: src/collision/collision.c (hunk 2 above)
Comment (m00nbyt3) [reply src/collision/collision.c:15]: True, solved

File: src/collision/collision.c (hunk 1 above)
Comment (m00nbyt3) [reply src/collision/collision.c:13]: True

File: src/collision/collision.c (hunk 3 above)
Comment (arikhativa) [reply src/collision/collision.c:47]: So i think the best way to do this is to create an enum
```
typedef enum e_conlision_stt
{
X_COLLITION,
Y_COLLITON,
NO_COLLITION
} t_conlision_stt
```

**Pull request closed without merging**